src/main/java/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -221,9 +223,16 @@ public void deleteStudyPermanently(String appId, String studyId) {
         checkNotNull(appId);
         checkNotNull(studyId);
         
+        Study existing = getStudy(appId, studyId, true);
+        
+        RequestContext context = RequestContext.get();
+        if (!CAN_DELETE_STUDY.contains(existing.getPhase()) && !context.isInRole(ADMIN)) {
+            throw new BadRequestException("Study cannot be deleted during phase " 
+                    + existing.getPhase().label());
+        }
+        
         // Throws exception if the element does not exist.
```

Comment:
Does this comment apply to the call got getStudy? Which got moved up.

Reply:
I don't know what "element" referred to. Removing.